### ISSUE TITLE
Can't call C# static function

### ISSUE THREAD

**Issue by CsloudX** [godotengine/godot#86256]
### Tested versions

v4.2.1.stable.mono.official [b09f793f5]

### System information

Godot v4.2.1.stable.mono - Windows 10.0.17763 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2060 (NVIDIA; 31.0.15.2849) - AMD Ryzen 9 5950X 16-Core Processor (32 Threads)

### Issue description

![image](https://github.com/godotengine/godot/assets/35988767/bddbbf8c-c7ab-4fb0-80ed-b70bb08d338e)

### Steps to reproduce

.

### Minimal reproduction project (MRP)

[CSStaticTest.zip](https://github.com/godotengine/godot/files/13709935/CSStaticTest.zip)

**Comment by zaevi** [comment]
This is a weird error when parsing GDScript, but the static function do works:

```GDScript
var klass = CsMain
var a = klass.F1()
print(a) # prints 123
```

**Comment by YuriSizov** [comment]
@CsloudX You didn't finish uploading the MRP. Please do not click away from the template field until it is done uploading next time you submit an issue.

For this one, could you please re-upload it?

**Comment by CsloudX** [comment]
Sorry about that, and I'm upload a new MRP.

**Comment by zaevi** [comment]
This is caused when lookup MethodInfo from CSharpScript:

https://github.com/godotengine/godot/blob/bf8dd73e9d88d6ab146506699f43230b12c6a2b0/modules/gdscript/gdscript_analyzer.cpp#L5016-L5018

https://github.com/godotengine/godot/blob/bf8dd73e9d88d6ab146506699f43230b12c6a2b0/core/object/object.h#L238

The `id` field is never used (at lease in C#) and is always 0. So the `info` always equals to empty `MethodInfo()`

**Comment by vnen** [comment]
It this fixed by https://github.com/godotengine/godot/pull/86259?

**Comment by zaevi** [comment]
> It this fixed by #86259?

![image](https://github.com/godotengine/godot/assets/12966814/575cdf24-4706-4175-8bcf-c5e14280104a)



It recognizes the static function now but gives an incorrect return type.

```csharp
[GlobalClass]
public partial class Main : Node
{
	public static int Add(int a, int b) => a + b; 

	// ...
}


```
